### ISSUE TITLE
Bad inlining of Infinity or NaN in PostgreSQL

### ISSUE THREAD

**Issue by smatytsin** [jOOQ/jOOQ#11701]
When I tried to get Inline SQL (`query.getSQL(ParamType.INLINED)`) for Postgres dialect query that contains NaN or +-Infinity values for bind variables, I have received incorrect syntax query:

```sql
select * from table_ex where dbl_prc_col = Infinity or dbl_prc_col = -Infinity or dbl_prc_col = NaN
```

this query should contains quotes around values +-Infinity and NaN:

```sql
select * from table_ex where dbl_prc_col = 'Infinity' or dbl_prc_col = '-Infinity' or dbl_prc_col = 'NaN'
```

For more details about such values for Floating-Point Types in Postgres see: https://www.postgresql.org/docs/8.2/datatype-numeric.html

**Comment by lukaseder** [comment]
Thanks a lot for your report. This looks similar to https://github.com/jOOQ/jOOQ/issues/6252. Will investigate soon

**Comment by lukaseder** [comment]
The `NaN` case is already implemented correctly, I believe? It should have been fixed since jOOQ 3.9 with https://github.com/jOOQ/jOOQ/issues/5249. How did you create your query?

**Comment by lukaseder** [comment]
Yeah, it's a duplicate of https://github.com/jOOQ/jOOQ/issues/6252. Will be fixed in jOOQ 3.15.0 and 3.14.9. Thanks again for your report.

**Comment by smatytsin** [comment]
Sorry, it was a copy-paste mistake. NaN is OK. Problem only with +-Infinity
